library/manageiq_custom_attributes.py

```diff
@@ -150,6 +150,15 @@ def update_custom_attribute(self, entity_type, entity_id, ca, ca_href):
         except Exception as e:
             self.module.fail_json(msg="Failed to update the custom attribute {ca_name}. Error: {error}".format(ca_name=ca['name'], error=e))
 
+    @staticmethod
+    def compare_custom_attributes(ca1, ca2):
+        if 'section' in ca1 and 'section' in ca2:
```

Comment:
I wanted to only compare with the section if the section is given in both. I thought of being backward compatible and assume the user wanted the default `metadata` or just stay with the current value for section but I see now that it will lead to an unpredictable behavior.